src/distPlus/symbolic/SymbolicDist.re

```diff
@@ -296,103 +313,407 @@ module GenericSimple = {
+    | `RenderedShape(sc, sd, 0.) => {
+        `RenderedShape(Distributions.Continuous.empty, Distributions.Discrete.empty, 0.)
+      }
+    | `RenderedShape(sc, sd, i) => {
+        // loop through all ys and divide them by i
+        let normalize = (s: DistTypes.xyShape): DistTypes.xyShape => {xs: s.xs, ys: s.ys |> E.A.fmap(y => y /. i)};
+
+        let scn = sc |> Distributions.Continuous.shapeMap(normalize);
+        let sdn = sd |> normalize;
+
+        `RenderedShape(scn, sdn, 1.)
+    }
+    | `Simple(d) => `Simple(d) // any kind of atomic dist should already be normalized -- TODO: THIS IS ACTUALLY FALSE! E.g. pointwise product of normal * normal
+    }
+  };
+
+  let evaluateTruncate = (et: nodeResult, xc: cutoffX, compareFunc: (float, float) => bool, n: int) => {
+    let cut = (s: DistTypes.xyShape): DistTypes.xyShape => {
+      let (xs, ys) = s.ys
+        |> Belt.Array.zip(s.xs)
+        |> E.A.filter(((x, y)) => compareFunc(x, xc))
+        |> Belt.Array.unzip
+
+      let cutShape: DistTypes.xyShape = {xs, ys};
+      cutShape;
+    };
+
+    switch (et) {
+      | `Simple(d) => {
+          let (sc, sd) = renderDistributionToXYShape(d, n);
+
+          let scc = sc |> Distributions.Continuous.shapeMap(cut);
+          let sdc = sd |> cut;
+
+          let newIntegral = 1.; // TODO
+
+          `RenderedShape(scc, sdc, newIntegral);
+      }
+      | `RenderedShape(sc, sd, i) => {
+          let scc = sc |> Distributions.Continuous.shapeMap(cut);
+          let sdc = sd |> cut;
+
+          let newIntegral = 1.; // TODO
+
+          `RenderedShape(scc, sdc, newIntegral);
+      }
+    }
+  };
+
+  let evaluateVerticalScaling = (et1: nodeResult, et2: nodeResult, n: int) => {
+    let scale = (i: float, s: DistTypes.xyShape): DistTypes.xyShape => {xs: s.xs, ys: s.ys |> E.A.fmap(y => y *. i)};
+
+    switch ((et1, et2)) {
+      | (`Simple(`Float(v)), `Simple(d))
+      | (`Simple(d), `Simple(`Float(v))) => {
+          let (sc, sd) = renderDistributionToXYShape(d, n);
+
+          let scc = sc |> Distributions.Continuous.shapeMap(scale(v));
+          let sdc = sd |> scale(v);
+
+          let newIntegral = v; // TODO
+
+          `RenderedShape(scc, sdc, newIntegral);
+      }
+      | (`Simple(`Float(v)), `RenderedShape(sc, sd, i))
+      | (`RenderedShape(sc, sd, i), `Simple(`Float(v))) => {
+          let scc = sc |> Distributions.Continuous.shapeMap(scale(v));
+          let sdc = sd |> scale(v);
+
+          let newIntegral = v; // TODO
+
+          `RenderedShape(scc, sdc, newIntegral);
+        }
+      | _ => `RenderedShape(Distributions.Continuous.empty, Distributions.Discrete.empty, 0.) // TODO: give warning
+    }
+  }
+
+  let renderNode = (et: nodeResult, n: int) => {
+    switch (et) {
+    | `Simple(d) => {
+          let (sc, sd) = renderDistributionToXYShape(d, n);
+          `RenderedShape(sc, sd, 1.0);
+      }
+    | s => s
+    }
+  }
+
+  let rec evaluateNode = (treeNode: distTree, n: int): nodeResult => {
```

Comment:
I think this function is complex enough to merit its' own module, maybe its own file. My impression is that this is the only function that calls many of its' subfunctions; that we won't want to call them externally. 

We may also want to use declaration files to make the API more explicit. 

Basically, this part is confusing from the outside (I'm looking at the module, figuring out what to directly do with it), and there are multiple strategies that can be used to help here.

Reply:
Also, I may prefer the name ``toNodeResult`` here, if that is what it's doing, and users need to be familiar with the ``nodeResult`` type.